src/components/App/index.js

```diff
@@ -1,56 +1,51 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
 import './App.css';
 import Header from '../Header';
 import Breeds from '../Breeds';
 import BreedPhotos from '../BreedPhotos';
+import { loadAllBreeds, selectBreed } from '../../actions/index';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
       breedName: '',
-      breeds: [],
-      photos: [],
-      hasErrors : false
     }
   }
 
   getSelectedBreed = (breedName) => {
     this.setState({ breedName: breedName })
-    this.selectBreed(breedName);
-  }
-
-  selectBreed = (breedName) => {
-    const photoUrl = `https://dog.ceo/api/breed/${ breedName }/images/random/2`
-    fetch(photoUrl)
-    .then(res => res.json())
-    .then(breed => this.setState({ photos: breed.message }))
-    .catch(() => this.setState({ hasErrors: true }))
+    this.props.selectBreed(breedName);
   }
 
   componentDidMount() {
-    const breedUrl = 'https://dog.ceo/api/breeds/list/all'
-    fetch(breedUrl)
-    .then(res => res.json())
-    .then(data => this.setState({ breeds: data.message }))
-    .catch(() => this.setState({ hasErrors: true }))
+    this.props.loadAllBreeds();
   }
 
   render() {
     return (
       <div>
         <Header />
         <Breeds 
-          breeds={ this.state.breeds } 
+          breeds={ this.props.breeds } 
           getSelectedBreed={ this.getSelectedBreed }
         />
         <BreedPhotos 
-          photos={ this.state.photos }
+          photos={ this.props.photos }
           breedName={ this.state.breedName }
         />
       </div>
     )
   }
 }
 
-export default App;
+const mapStateToProps = (state) => {
+  return ({
+    breeds: state.breeds,
+    photos: state.photos
+  })
+}
```

Comment:
Fun fact: Arrow functions will return a value or execute a function if the arrow points at `{`.  So what do you do when you want to just return an object?

You can wrap the object in parentheses!

```jsx
const mapStateToProps = (state) => ({
  breeds: state.breeds,
  photos: state.photos
});

Reply:
That's right! Thank you for the feedback.